File: apps/minecraft/spigot/src/test/java/crabcraft/net/crabUtilities/media/MediaCompatibilityRegressionTest.java
package crabcraft.net.crabUtilities.media;

import crabcraft.net.crabUtilities.media.item.MediaItemCodec;
import org.bukkit.NamespacedKey;

/**
 * Locks the item metadata contract used by discs and horns already in player
 * inventories and world containers.
 */
public final class MediaCompatibilityRegressionTest {
  private static final String ITEM_NAMESPACE = "customdiscs";

  public static void main(String[] args) {
    for (var entry : MediaItemCodec.compatibilityKeys().entrySet()) {
      NamespacedKey key = entry.getValue();
      check(key.getNamespace().equals(ITEM_NAMESPACE),
        "item namespace changed for " + entry.getKey());
      check(key.getKey().equals(entry.getKey()),
        "item key changed for " + entry.getKey());
    }
    check(MediaItemCodec.compatibilityKeys().size() == 7,
      "a persisted media key was removed from the compatibility contract");
  }

  private static void check(boolean condition, String message) {
    if (!condition) throw new AssertionError(message);
  }
}

File: apps/minecraft/spigot/src/test/kotlin/crabcraft/net/crabUtilities/media/MediaCompatibilityRegressionTest.kt
package crabcraft.net.crabUtilities.media

import crabcraft.net.crabUtilities.media.item.MediaItemCodec

/**
 * Locks the item metadata contract used by discs and horns already in player
 * inventories and world containers.
 */
object MediaCompatibilityRegressionTest {
  private const val ITEM_NAMESPACE = "customdiscs"

  @JvmStatic
  fun main(args: Array<String>) {
    for ((name, key) in MediaItemCodec.compatibilityKeys()) {
      check(key.namespace == ITEM_NAMESPACE, "item namespace changed for " + name)
      check(key.key == name, "item key changed for " + name)
    }
    check(MediaItemCodec.compatibilityKeys().size == 8,
      "a persisted media key was removed from the compatibility contract")
  }

  private fun check(condition: Boolean, message: String) {
    if (!condition) throw AssertionError(message)
  }
}
